src/palimpzest/sets.py

```diff
@@ -293,6 +293,27 @@ def add_columns(self, udf: Callable,
             cache=self._cache,
         )
 
+    def map(self, udf: Callable, depends_on: str | list[str] | None = None) -> Dataset:
+        """
+        Apply a UDF map function.
+
+        Examples:
+            map(udf=clean_column_values)
+        """
+        if udf is None:
+            raise ValueError("`udf` must be provided for map.")
+
+        if isinstance(depends_on, str):
```

Comment:
Seems like we don't use the depeneds_on parameter in other parts of the code?

Reply:
You're technically correct; in the future if / when we have more transformation rules for pushing maps and converts around the plan, then this could be useful -- but for now I will take it out.